infra/modules/service/main.tf

```diff
@@ -303,6 +335,84 @@ data "aws_iam_policy_document" "task_executor" {
   }
 }
 
+# ECS task role and policy
+# Only defined if needed
+resource "aws_iam_role" "task" {
+  count              = local.define_ecs_task_role ? 1 : 0
+  name               = local.task_role_name
+  assume_role_policy = data.aws_iam_policy_document.ecs_assume_task_role.json
+}
+
+data "aws_iam_policy_document" "ecs_assume_task_role" {
+  statement {
+    sid = "ECSTask"
+    actions = [
+      "sts:AssumeRole"
+    ]
+    principals {
+      type        = "Service"
+      identifiers = ["ecs-tasks.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_policy" "task" {
+  count       = local.define_ecs_task_role ? 1 : 0
+  name        = "${var.service_name}-task-role-policy"
+  description = "A policy for ECS task"
+  policy      = data.aws_iam_policy_document.task.json
+}
+
+# Link access policies to the ECS task role.
+resource "aws_iam_role_policy_attachment" "task" {
+  count      = local.define_ecs_task_role ? 1 : 0
+  role       = aws_iam_role.task[0].name
+  policy_arn = aws_iam_policy.task[0].arn
+}
+
+data "aws_iam_policy_document" "task" {
+  # Allow ECS to access SSM Messages so that ECS Exec works
+  # See https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-exec.html
```

Comment:
Are we using ECS exec in production?

Reply:
I'm not planning on it, but I needed it for debugging and left the functionality to turn it on/off with a variable `enable_exec` in this PR. `var.enable_exec` defaults to `false`